### PR TITLE
fix issue #431 where test coverage deps were excluded

### PR DESCRIPTION
**Pull request by isao**
change "tests" from .npmignore to "/tests" as these
former case excluded things required for mojito test,
specifically lib/tests/harness/lib/yuitest/java

TODO use coverage jars in yuitest npm pkgs or similar
(though the don't appear in npm yui or yuitest yet)?

also ignore "node_modules" in top level or in examples/*
only (test fixtures need to contain node_modules dirs

### CODE REVIEW

**Comment by mridgway** [comment]
I think we should exclude all `tests` as it was before and then add back in the jars that are needed (!lib/tests/harness). The tests themselves are not needed by a downstream application, so we should exclude them.

**Comment by isao** [comment]
top level (arrow) tests are excluded

but we need fw tests to support `mojito test` command per the docs

File: .gitignore [hunk 1]
```diff
@@ -1,12 +1,12 @@
 .DS_Store
-node_modules
-!tests/fixtures/packages/node_modules
-!tests/fixtures/packages/node_modules/*/node_modules
```
Comment (drewfish) [inline .gitignore:4]: So `tests/fixtures/packages/node_modules` (and children) are definitely needed in order to do proper testing in the resource store.  Does the new .gitignore support that (it's hard for me to tell from looking at the rules).

**Comment by isao** [comment]
BTW, we should get rid of lib/tests/harness/lib/yuitest/ in favor of npm packages with https://npmjs.org/package/yui-coverage

**Comment by mridgway** [comment]
@isao Agreed. I think we should evaluate that as we make `mojito test` work with Arrow.

File: .gitignore (hunk 1 above)
Comment (isao) [inline .gitignore:4]: yes, I know, I said as much this morning.. :)

`./node_modules` and `./examples/**/node_modules` are _excluded_ and everything else is allowed

Comment (mridgway) [inline .gitignore:4]: Yeah, I'd prefer to be explicit about what we want to keep as it was before.

Comment (isao) [inline .gitignore:4]: yes, sorry for the funny diff.. just look at the full file.

I sorted to compare with `.npmignore`

**Pull request merged**